Tests/L0/Gradle/_suite.ts

```diff
@@ -13,36 +13,62 @@ function setResponseFile(name: string) {
     process.env['MOCK_RESPONSES'] = path.join(__dirname, name);
 }
 
-function setDefaultInputs(tr:TaskRunner):TaskRunner {
+function setDefaultInputs(tr: TaskRunner, enableSonarQubeAnalysis: boolean): TaskRunner {
     tr.setInput('wrapperScript', 'gradlew'); // Make that checkPath returns true for this filename in the response file
     tr.setInput('options', '');
     tr.setInput('tasks', 'build');
     tr.setInput('javaHomeSelection', 'JDKVersion');
     tr.setInput('jdkVersion', 'default');
     tr.setInput('publishJUnitResults', 'true');
     tr.setInput('testResultsFiles', '**/build/test-results/TEST-*.xml');
+
+    if (enableSonarQubeAnalysis) {
+        tr.setInput('sqAnalysisEnabled', 'true');
+        tr.setInput('sqConnectedServiceName', 'ID1');
+        tr.setInput('sqProjectName', 'test_sqProjectName');
+        tr.setInput('sqProjectKey', 'test_sqProjectKey');
+        tr.setInput('sqProjectVersion', 'test_sqProjectVersion');
+    }
+    else {
+        tr.setInput('sqAnalysisEnabled', 'false');
+    }
+
     return tr;
 }
 
-describe('gradle Suite', function() {
+function setResponseAndBuildVars(initialResponseFile:string, finalResponseFile:string, envVars: Array<[string,string]> ) {
```

Comment:
Yes, I think those JSON files make the test very difficult to read, especially as you cannot add comments in Json. This way the intent of the tests is much clearer.